de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolAdmin/service/impl/ToolAdminServiceFacadeImpl.java

```diff
@@ -93,13 +104,11 @@ public void publishTool(Tool tool, AsyncCallback<Void> callback) {
         String address = TOOLS_ADMIN + "/" + tool.getId() + "/publish";
 
         //null out values not needed by service
-        Splittable s = AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(tool));
-        Splittable.NULL.assign(s, "is_public");
-        Splittable.NULL.assign(s, "permission");
+        Splittable encode = AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(tool));
+        nullUnwantedValues(encode);
 
         ServiceCallWrapper wrapper = new ServiceCallWrapper(POST,
-                                                            address,
-                                                            s.getPayload());
+                                                            address, encode.getPayload());
```

Comment:
Minor formatting issue: all these params should go on one line or each on their own line.